CRM/Report/Form.php

```diff
@@ -577,6 +577,11 @@ public function preProcessCommon() {
       else {
         $this->_formValues = NULL;
       }
+      // CRM-19330 Unprivileged users don't get the task in the form values,
+      // but they appear in the post
+      if (!$this->_formValues['task'] && filter_input(INPUT_POST, 'task') !== NULL) {
+        $this->_formValues['task'] = filter_input(INPUT_POST, 'task');
+      }
```

Comment:
@spalmstr If `task` appears in post, $this->_ouputMode would be correctly set using `$_REQUEST` or `$this->_params` in setOutputMode() called below. Do we need this part as I was able to perform all the remaining task (print, pdf, export) without this condition too.